src/assembler.jl

```diff
@@ -89,71 +100,92 @@ Assembles the element residual `ge` into the global residual vector `g`.
     end
 end
 
-abstract type AbstractSparseAssembler end
-
 """
     matrix_handle(a::AbstractSparseAssembler)
     vector_handle(a::AbstractSparseAssembler)
 
-Return a reference to the underlying matrix/vector of the assembler.
+Return a reference to the underlying matrix/vector of the assembler used during
+assembly operations.
 """
 matrix_handle, vector_handle
 
-struct AssemblerSparsityPattern{Tv,Ti} <: AbstractSparseAssembler
-    K::SparseMatrixCSC{Tv,Ti}
+"""
+    get_matrix(a::AbstractSparseAssembler)
+    get_vector(a::AbstractSparseAssembler)
+
+Return a reference to the matrix into which we assemble.
+The returned object might differ from [`matrix_handle`](@ref).
+"""
+get_matrix, get_vector
+
+get_vector(a::AbstractSparseAssembler) = a.f
+get_matrix(a::AbstractSparseAssembler) = a.K
+
+"""
+Assembler for sparse matrix with CSC storage type.
+"""
+struct CSCAssembler{Tv,Ti,MT<:AbstractSparseMatrixCSC{Tv,Ti}} <: AbstractSparseCSCAssembler
+    K::MT
     f::Vector{Tv}
     permutation::Vector{Int}
     sorteddofs::Vector{Int}
 end
```

Comment:
Since we are renaming and introducing new types here, would it make sense to get rid of the symmetric type, seems like some code could be replaced which AFAIU is only used to get a bool when calling `_assemble!`? E.g. 
```suggestion
struct CSCAssembler{Tv, Ti, MT} <: AbstractSparseCSCAssembler
    K::MT
    f::Vector{Tv}
    permutation::Vector{Int}
    sorteddofs::Vector{Int}
    function CSCAssembler(K::Union{DT, Symmetric{Tv, TT}}, f::Vector{Tv}, permutation::Vector{Int}, sorteddofs::Vector{Int}) where {Tv, Ti, DT<:AbstractSparseMatrixCSC{Tv,Ti}}
        return new{Tv, Ti, typeof(K)}(K, f, permutation, sorteddofs)
    end
end
symmetric_pattern(::CSCAssembler) = false
symmetric_pattern(::CSCAssembler{<:Any, <:Any, <:Symmetric}) = true
```

with 
```julia
matrix_handle(K::Symmetric) = K.data
matrix_handle(K::AbstractMatrix) = K

matrix_handle(a::AbstractSparseAssembler) = matrix_handle(get_matrix(a))
```
alternatively 
```julia
matrix_handle(a::AbstractSparseAssembler) = symmetric_pattern(a) ? get_matrix(a).data : get_matrix(a)
```

Reply:
I think @fredrikekre wanted to remove the symmetric assembler type anyways, and with this we maintain all functionality AFAIU.